staging/src/k8s.io/apiserver/pkg/registry/rest/resttest/resttest.go

```diff
@@ -917,20 +919,24 @@ func (t *Tester) testDeleteWithResourceVersion(obj runtime.Object, createFn Crea
 
 	foo := obj.DeepCopyObject()
 	t.setObjectMeta(foo, t.namer(1))
-	objectMeta := t.getObjectMetaOrFail(foo)
-	objectMeta.SetResourceVersion("RV0000")
 	if err := createFn(ctx, foo); err != nil {
 		t.Errorf("unexpected error: %v", err)
 	}
-	opts.Preconditions = metav1.NewRVDeletionPrecondition("RV1111").Preconditions
+	newObj, err := getFn(ctx, foo)
+	if err != nil {
+		t.Errorf("unexpected error: %v", err)
+	}
+	objectMeta := t.getObjectMetaOrFail(newObj)
+
+	opts.Preconditions = metav1.NewRVDeletionPrecondition("wrongVersion").Preconditions
```

Comment:
I suspect "wrongVersion" will produce an error other than a conflict error, since it won't parse as a valid resourceVersion on the server-side

Reply:
@liggitt In fact, it raised a conflict error: 

```
Message: "Operation cannot be fulfilled on Noxu.mygroup.example.com "foo1": the ResourceVersion in the precondition (wrongVersion) does not match the ResourceVersion in record (10). The object might have been modified"
Reason: "Conflict"
Code: 409

```

Reply:
ok, great